src/js/components/deployment-view/index.tsx

```diff
@@ -49,38 +48,26 @@ class DeploymentView extends React.Component<Props, void> {
   }
 
   public componentWillMount() {
-    const { loadPreviewAndComments, isUserLoggedIn } = this.props;
-    const { deploymentId, projectId, branchId, token } = this.props.params;
-
-    if (deploymentId) {
-      loadPreviewAndComments(deploymentId, 'deployment', token, isUserLoggedIn);
-    } else if (projectId) {
-      loadPreviewAndComments(projectId, 'project', token, isUserLoggedIn);
-    } else if (branchId) {
-      loadPreviewAndComments(branchId, 'branch', token, isUserLoggedIn);
-    } else {
+    const { loadPreviewAndComments, isUserLoggedIn, params: { entityType, token, id } } = this.props;
```

Comment:
`componentWillMount` and `componentWillReceiveProps` are so similar that extracting a common function might be worthwhile